client/my-sites/people/people-list-item/index.tsx

```diff
@@ -0,0 +1,227 @@
+import { createNotice, removeNotice } from 'calypso/state/notices/actions';
+import { NoticeStatus } from 'calypso/state/notices/types';
+import { AppState } from 'calypso/types';
+import { Invite } from '../team-invites/types';
+import type { Member, SiteDetails } from '@automattic/data-stores';
+import './style.scss';
+
+interface PeopleListItemProps {
+	site?: SiteDetails | null;
+	invite?: Invite;
+	clickableItem?: boolean;
+	user?: Member;
+	isSelectable?: boolean;
+	onRemove?: () => void;
+	type?: string;
+}
+
+interface sendInvitesResponse {
+	success: boolean;
+	errors: Array< string >;
+	email_or_username: string;
+}
+
+const PeopleListItem: React.FC< PeopleListItemProps > = ( {
+	site,
+	invite,
+	user,
+	isSelectable,
+	onRemove,
+	type,
+	clickableItem = true,
+}: PeopleListItemProps ) => {
+	const siteId = site && site?.ID;
+	const inviteKey = invite?.key;
+	const dispatch = useDispatch();
+	const translate = useTranslate();
+
+	const inviteWasDeleted = useSelector( ( state: AppState ) => {
+		return siteId && inviteKey && didInviteDeletionSucceed( state, siteId, inviteKey );
+	} );
+
+	const { isPending: isSubmittingInvites, mutateAsync: sendInvites } = useSendInvites(
+		siteId as number
+	);
+
+	const navigateToUser = () => {
+		window.scrollTo( 0, 0 );
+		dispatch(
+			composeAnalytics( recordGoogleEvent( 'People', 'Clicked User Profile From Team List' ) )
+		);
+	};
+
+	const userHasPromoteCapability = () => {
+		return site && site.capabilities && site.capabilities.promote_users;
+	};
+
+	const canLinkToProfile = () => {
+		return user && user.roles && site && site.slug && userHasPromoteCapability() && ! isSelectable;
+	};
+
+	const canLinkToSubscriberProfile = () => {
+		return site && site.slug && user && user.ID;
+	};
+
+	const canReceiveInvite = () => {
+		return (
+			user &&
+			user.roles &&
+			user.email &&
+			! user.linked_user_ID &&
+			site &&
+			site.slug &&
+			! isSelectable
+		);
+	};
+
+	const displayNotice = (
+		message: string,
+		noticeType: NoticeStatus = 'is-success',
+		duration: undefined | number | null = 3000,
+		additionalOptions: { button?: string; id?: string; onClick?: () => void } = {}
+	) => {
+		const { notice } = dispatch(
+			createNotice( noticeType, message, { duration, ...additionalOptions } )
+		);
+		return {
+			removeNotice: () => dispatch( removeNotice( notice.noticeId ) ),
+		};
+	};
+
+	const onSendInvite = async ( event: React.MouseEvent< HTMLButtonElement, MouseEvent > ) => {
+		// Prevents navigation to invite-details screen and onClick event.
+		event.preventDefault();
+		event.stopPropagation();
+
+		if ( isSubmittingInvites || ! user || ! user.roles ) {
+			return null;
+		}
+		const { email } = user;
+		const userRoles = getRole( user );
+		const response = ( await sendInvites( [
+			{ email_or_username: email as string, role: userRoles },
+		] ) ) as [ sendInvitesResponse ];
+
+		const success = response && response[ 0 ].success;
```

Comment:
Can add safety checks here?
Also we need to manage errors, and maybe log the invite error event. WDYT?